pkg/controller/clusterinstallation/mattermost.go

```diff
@@ -59,6 +59,21 @@ func (r *ReconcileClusterInstallation) checkMattermostService(mattermost *matter
 		update = true
 	}
 
+	// If we are using the loadBalancer the ClusterIp is immutable
```

Comment:
I don't think we need this section at all. The following code block does a `reflect.DeepEqual` on the `Spec` and then replaces the whole thing if anything is different.

Did you run into that not doing the trick in your manual testing?

Reply:
if use that without this change will be in an infinite loop updating the service

Reply:
Hmm, that's interesting. We can go with this and review it later if needed.